### PR TITLE
Reporter implementation

### PR DESCRIPTION
**Pull request by RamLavi**
Add the Reporter implementation, based on the VM latency checkup's reporter [1].
Add the ability to report checkup status to the user.

[1] https://github.com/kiagnose/kiagnose/tree/95d8c7995fabbb7be11f7cde0eca57dc01e222bb/checkups/kubevirt-vm-latency/vmlatency/internal/reporter

### CODE REVIEW

**Comment by RamLavi** [comment]
Manually tested on an OpenShift 4.11 cluster.

```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1673987200

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [15.365 seconds]
------------------------------

Ran 1 of 1 Specs in 15.369 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (15.37s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	15.400s
```

File: pkg/internal/reporter/reporter.go [hunk 1]
```diff
@@ -19,15 +19,54 @@
...
+
+	kreporter "github.com/kiagnose/kiagnose/kiagnose/reporter"
+
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/status"
+)
+
+const (
+	DropRate             = "dropRate"
+	TrafficGeneratorNode = "trafficGeneratorNode"
+	DPDKVMNode           = "DPDKVMNode"
```
Comment (orelmisan) [inline pkg/internal/reporter/reporter.go:35]: Please consider adding the `Key` suffix to these constants.

File: pkg/internal/reporter/reporter_test.go [hunk 2]
```diff
@@ -20,16 +20,150 @@
...
 }
+
+func TestReportShouldFailWhenCannotUpdateConfigMap(t *testing.T) {
+	// ConfigMap does not exist
+	fakeClient := fake.NewSimpleClientset()
+
+	testReporter := reporter.New(fakeClient, testNamespace, testConfigMapName)
+
+	assert.ErrorContains(t, testReporter.Report(status.Status{}), "not found")
+}
```
Comment (orelmisan) [inline pkg/internal/reporter/reporter_test.go:60]: Please consider moving this test after the success scenarios.

Comment (RamLavi) [reply pkg/internal/reporter/reporter_test.go]: Done

File: pkg/internal/reporter/reporter.go (hunk 1 above)
Comment (RamLavi) [reply pkg/internal/reporter/reporter.go]: Done

**Pull request merged**